lib/spack/spack/test/cmd/compiler.py

```diff
@@ -103,3 +104,92 @@ def test_compiler_add(
     new_compiler = new_compilers - old_compilers
     assert any(c.version == spack.version.Version(mock_compiler_version)
                for c in new_compiler)
+
+
+@pytest.fixture
+def clangdir(tmpdir):
+    """Create a directory with some dummy compiler scripts in it.
+
+    Scripts are:
+      - clang
+      - clang++
+      - gfortran-8
+
+    """
+    with tmpdir.as_cwd():
+        with open('clang', 'w') as f:
+            f.write("""\
+#!/bin/sh
+if [ "$1" = "--version" ]; then
+    echo "Apple clang version 11.0.0 (clang-1100.0.33.16)"
+    echo "Target: x86_64-apple-darwin18.7.0"
+    echo "Thread model: posix"
+    echo "InstalledDir: /dummy"
+else
+    echo "clang: error: no input files"
+fi
+""")
+        shutil.copy('clang', 'clang++')
+
+        with open('gfortran-8', 'w') as f:
+            f.write("""\
+#!/bin/sh
+if [ "$1" = "--version" ]; then
+    echo "GNU Fortran (GCC) 8.4.0 20120313 (Red Hat 8.4.0-1)"
+    echo "Copyright (C) 2010 Free Software Foundation, Inc.
+else
+    echo "gfortran-8: fatal error: no input files"
+    echo "compilation terminated."
+fi
+""")
+        os.chmod('clang', 0o700)
+        os.chmod('clang++', 0o700)
+        os.chmod('gfortran-8', 0o700)
+
+    yield tmpdir
```

Comment:
I think this can be solved by adding:
```suggestion
    yield tmpdir
    
    spack.compiler.get_compiler_version_output.cache.clear()
```
here and in `mock_compiler_dir` plus the necessary import at the top of the file.